kyuubi-server/src/main/scala/org/apache/kyuubi/engine/hive/HiveProcessBuilder.scala

```diff
@@ -81,16 +81,25 @@ class HiveProcessBuilder(
     extraCp.foreach(classpathEntries.add)
     if (extraCp.isEmpty) {
       warn(s"The conf of kyuubi.engine.hive.extra.classpath is empty.")
-      mainResource.foreach { path =>
-        val devHadoopJars = Paths.get(path).getParent
-          .resolve(s"scala-$SCALA_COMPILE_VERSION")
-          .resolve("jars")
-        if (!Files.exists(devHadoopJars)) {
-          throw new KyuubiException(s"The path $devHadoopJars does not exists. Please set " +
-            s"kyuubi.engine.hive.extra.classpath for configuring location of " +
-            s"hadoop client jars, etc")
+      val distJars = new File(s"${env.get(KYUUBI_HOME).orNull}${File.separator}jars")
```

Comment:
Kyuubi Server lib is designed isolated with engine jars, if we mix them it's hard to maintain in the future

Reply:
Could I package `hadoop-client*jar` and `common-collection-*jar` into `dist/externals/engines/hive/jars`, and load this path into classpath? Or, I enable `HIVE_HADOOP_CLASSPATH` to let users set these jars?

Reply:
> Could I package hadoop-client*jar and common-collection-*jar into dist/externals/engines/hive/jars

I have done this before, but this 1) make the binary distribution too large. 2) it may become a blocker for multiple hive version support

kyuubi.engine.hive.extra.classpath will do the same thing as HIVE_HADOOP_CLASSPATH

the thing is the AT-LEAST classpath of a hive engine is made by
- hive-sql-engine.jar
- HIVE_HOME/libs with a hive distribution
- hadoop client jars

And here we mainly focus on the **hadoop client jars** and the deps. 

- Is it possible to make it transparent for users？
   - a fat jar within  hive-sql-engine.jar?
   - make a separate folder and deliver it with the binary release?
- If not, how to make it easier for user to understand and use
   - documentation is required
   - HIVE_HADOOP_CLASSPATH or kyuubi.engine.hive.extra.classpath, or both?
      - what are their priorities?
      - shall we raise an error or not?

Reply:
Another thing that we shall not add the Hadoop client jars by default is
- both hive distribution and the hive-sql-engine.jar are not created for specific Hadoop versions, users can set accordingly. If we don't propagate them, class not found exception; if we do, compatibility issue lurks.

Reply:
> If not, how to make it easier for user to understand and use

I second this. I will achieve loading both HIVE_HADOOP_CLASSPATH and `kyuubi.engine.hive.extra.classpath`.
HIVE_HADOOP_CLASSPATH is required. And I will explain how to set HIVE_HADOOP_CLASSPATH in doc.